src/js/views/patients/schedule/schedule_views.js

```diff
@@ -368,17 +371,13 @@ const ScheduleListView = CollectionView.extend({
   viewComparator(viewA, viewB) {
     return alphaSort('asc', viewA.model.get('date'), viewB.model.get('date'));
   },
-  viewFilter(view) {
-    if (this.state.get('searchQuery')) {
-      return !view.isEmpty();
-    }
+  filterOutEmptyDays() {
+    this.setFilter(function({ children, model }) {
+      // 'null' string is a key from groupBy
+      if (!children.length || model.get('date') === 'null') return false;
 
-    // 'null' string is a key from groupBy
-    if (view.model.get('date') === 'null') {
-      return false;
-    }
-
-    return true;
+      return true;
+    });
```

Comment:
Things appear to be working with the schedule now. Not sure if this is the best way to do it though 🤷

Reply:
going to look at this more tomorrow to wrap my 🧠 around it.

Reply:
Having a hard time with this one.  This was previously removing the collection that has no "day" because there's no where to put them..  _but_ it currently removes days that have no children with find in list.. so what about this PR requires this?

Reply:
I think the issue is that this filter first runs prior to the children rendering, so they're always empty.
This avoids this:
```js
  viewFilter(view) {
    if (this.isAttached() && this.state.get('searchQuery')) {
      return !view.isEmpty();
    }

    // 'null' string is a key from groupBy
    if (view.model.get('date') === 'null') {
      return false;
    }

    return true;
  },
```
but there might be a flash of content if there's lots of stuff.. still though.. it kinda has to render to be applied.

Thoughts?

Reply:
Ah `this.isAttached()` is what I was looking for... nice. That works well. I just pushed that change

But ya that's the problem I was trying to solve with the `filterOutEmptyDays()` function. To have the view filter applied after the children were rendered. Otherwise, the view filter was clearing out all the days even when it shouldn't.

Ya I don't think there's a way to avoid the layout flash/shift.

Reply:
Ah this is essentially what you suggested in the earlier comments.. for whatever reason those are showing up on an "outdated" commit so I missed it till I came back to the main page 🤦

Reply:
Haha all good!